Traceback (most recent call last):
  File "C:\Users\T260366\OneDrive - PPG Industries, Inc\Github\pyppg-handbook\venv\Lib\site-packages\jupyter_cache\executors\utils.py", line 58, in single_nb_execution
    executenb(
  File "C:\Users\T260366\OneDrive - PPG Industries, Inc\Github\pyppg-handbook\venv\Lib\site-packages\nbclient\client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\T260366\OneDrive - PPG Industries, Inc\Github\pyppg-handbook\venv\Lib\site-packages\jupyter_core\utils\__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.1520.0_x64__qbz5n2kfra8p0\Lib\asyncio\base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "C:\Users\T260366\OneDrive - PPG Industries, Inc\Github\pyppg-handbook\venv\Lib\site-packages\nbclient\client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "C:\Users\T260366\OneDrive - PPG Industries, Inc\Github\pyppg-handbook\venv\Lib\site-packages\nbclient\client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "C:\Users\T260366\OneDrive - PPG Industries, Inc\Github\pyppg-handbook\venv\Lib\site-packages\nbclient\client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------

company = mr.MultiSelect(value=emp_df.Company_Name.unique(), 
                         choices=emp_df.Company_Name.unique(),
                         label="Select Companies")


credits_filter = mr.Slider(value=1, min=emp_df.Credits.min(), 
                           max=emp_df.Credits.max(), label="Credits Filter", step=1)

new_df = emp_df[(emp_df.Company_Name.isin(company.value)) & (emp_df.Credits>=int(credits_filter.value))]
mr.Md(f"The DataFrame has {new_df.shape[0]} rows and {new_df.shape[1]} columns.")

if len(new_df):
    fig, ax = plt.subplots(1, 2, figsize = (16, 9))
    sns.scatterplot(data = new_df, ax = ax[0], x = "Employee_Rating", y = "Employee_Salary")
    sns.countplot(x = new_df.Employment_Status, ax = ax[1])
    plt.show();
------------------


[1;31m---------------------------------------------------------------------------[0m
[1;31mNameError[0m                                 Traceback (most recent call last)
Cell [1;32mIn[4], line 1[0m
[1;32m----> 1[0m company [38;5;241m=[39m mr[38;5;241m.[39mMultiSelect(value[38;5;241m=[39m[43memp_df[49m[38;5;241m.[39mCompany_Name[38;5;241m.[39munique(), 
[0;32m      2[0m                          choices[38;5;241m=[39memp_df[38;5;241m.[39mCompany_Name[38;5;241m.[39munique(),
[0;32m      3[0m                          label[38;5;241m=[39m[38;5;124m"[39m[38;5;124mSelect Companies[39m[38;5;124m"[39m)
[0;32m      6[0m credits_filter [38;5;241m=[39m mr[38;5;241m.[39mSlider(value[38;5;241m=[39m[38;5;241m1[39m, [38;5;28mmin[39m[38;5;241m=[39memp_df[38;5;241m.[39mCredits[38;5;241m.[39mmin(), 
[0;32m      7[0m                            [38;5;28mmax[39m[38;5;241m=[39memp_df[38;5;241m.[39mCredits[38;5;241m.[39mmax(), label[38;5;241m=[39m[38;5;124m"[39m[38;5;124mCredits Filter[39m[38;5;124m"[39m, step[38;5;241m=[39m[38;5;241m1[39m)
[0;32m      9[0m new_df [38;5;241m=[39m emp_df[(emp_df[38;5;241m.[39mCompany_Name[38;5;241m.[39misin(company[38;5;241m.[39mvalue)) [38;5;241m&[39m (emp_df[38;5;241m.[39mCredits[38;5;241m>[39m[38;5;241m=[39m[38;5;28mint[39m(credits_filter[38;5;241m.[39mvalue))]

[1;31mNameError[0m: name 'emp_df' is not defined

